generic/image.c

```diff
@@ -2103,6 +2103,114 @@ int image_(Main_rgb2y)(lua_State *L) {
   return 0;
 }
 
+static inline void image_(drawPixel)(THTensor *output, int y, int x,
+				     int cr, int cg, int cb) {
+#ifdef TH_REAL_IS_BYTE
+	  THTensor_(set3d)(output, 0, y, x, cr);
+	  THTensor_(set3d)(output, 1, y, x, cg);
+	  THTensor_(set3d)(output, 2, y, x, cb);
+#else
+	  THTensor_(set3d)(output, 0, y, x, cr / 255);
+	  THTensor_(set3d)(output, 1, y, x, cg / 255);
+	  THTensor_(set3d)(output, 2, y, x, cb / 255);
+#endif
+}
+static inline void image_(drawChar)(THTensor *output, int x, int y, char c, int size,
+			     int cr, int cg, int cb,
+			     int bg_cr, int bg_cg, int bg_cb) {
+  long channels = output->size[0];
+  long height = output->size[1];
+  long width  = output->size[2];
+
+  /* out of bounds condition, return without drawing */
+  if((x >= width)            || // Clip right
+     (y >= height)           || // Clip bottom
+     ((x + 6 * size - 1) < 0) || // Clip left
+     ((y + 8 * size - 1) < 0))   // Clip top
+    return;
+
+  for(char i = 0; i < 6; i++ ) {
+    unsigned char line;
+    if (i < 5) {
+      line = *(const unsigned char *)(image_ada_font+(c*5) + i);
+    } else {
+      line = 0x0;
+    }
+    for(char j = 0; j < 8; j++, line >>= 1) {
+      if(line & 0x1) {
+	if (size == 1) {
```

Comment:
It's really minor but here and on the following lines you have some tabs instead of spaces for indenting (same in some other places in this PR like `drawPixel`, etc). Would be nice to switch back to spaces, right?